server/channels/app/plugin_db_driver.go

```diff
@@ -21,7 +22,7 @@ import (
 type DriverImpl struct {
 	s       *Server
 	connMut sync.RWMutex
-	connMap map[string]*sql.Conn
+	connMap map[string]*connMeta
```

Comment:
I first had a sense that the top level key should be the pluginID, but I think it may require some re-writing in the code.

Reply:
Well, we need connID at the top level for sure for easy access to the conn struct for query operations. What I was considering was to add another map with pluginID as the key. But then I realized that the only case to support for was plugin shutdown which is not in the critical path. So I think it's okay to have a complexity of O(n) to shutdown a plugin, and not incur additional memory cost.